tests/jballerina-unit-test/src/test/resources/test-src/object/object_functions_with_default_parameters_negative.bal

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public class Class1 {
+    int i = 0;
+
+    public function foo(int arg1 = i) { // error
+        int _ = self.i;
+    }
+}
+
+public class Class2 {
+    int i = 0;
+
+    public function foo(int i = i) { // error
+        
+    }
+}
+
+class Class3 {
+    int i = 0;
+    string s = "";
+
+    public function foo(int arg1 = i, string arg2 = s) { // errors
+        
+    }
+}
+
+class Class4 {
+    int i = 0;
+    string s = "";
+    string|int u;
+
+    function init() {
+        self.u = 2;
+    }
+
+    public function foo(int arg1 = i, string arg2 = s, string|int arg3 = u) { // errors
+        
+    }
+}
+
+public class Class5 {
+    int i = 0;
+
+    public function foo(int arg1 = i) { // error
+        int _ = self.i;
+    }
+
+    public function bar() {
+        self.foo(2);
+    }
+}
+
+public class Class6 {
+    int i = 0;
+
+    public function foo(int arg1 = i) { // error
+        object {} _ = object {
+            int i = 0;
+
+            function foo(int arg2 = i) { // error
```

Comment:
We should be able to refer to fields from the enclosing class right? Shall add a test case for that also, but with different field name?

Reply:
Do you mean an example like this?

```ballerina
public class Class {
    int i = 0;

    public function foo(int arg1) {
        object {} _ = object {
            function foo(int arg2) {
                int _ = self.i;
            }
        };
    }
}
```

Reply:
No i mean, can we refer to the fields in outer most class as defaultable values to the functions in anonymous objects?

Reply:
I think it is already covered

Reply:
```ballerina
public class Class {
    int i = 0;

    public function foo(int arg1) {
        object {} _ = object {
            function foo(int arg2 =  self.i) {
                int _ = self.i;
            }
        };
    }
}
```

like this?

Reply:
This example does not work as expected. 
The reason is that method `foo` of the object can see the self of class `Class`. Therefore `int arg2 = self.i` is not identified as an error. 
I think this is not related to this PR. @gimantha shall we fix it separately?

Reply:
Lets create a separate issue then?